tests/engagement/test_churn.py

```diff
@@ -383,3 +384,68 @@ def case(name, expect, desktop, mobile, configured):
 
     for test in test_func:
         test(df)
+
+
+def test_attribution_from_new_profile(effective_version,
+                                      generate_main_summary_data,
+                                      generate_new_profile_data):
+    main_summary = generate_main_summary_data([
+        {'client_id': '1', 'attribution': {'source': 'mozilla.org'}},
+        {'client_id': '3', 'attribution': None},
+        {'client_id': '4', 'attribution': None},
+    ])
+
+    def update_attribution(attribution):
+        # only updates the attribution section in the environment
+        env = copy.deepcopy(data.new_profile_sample['environment'])
+        env['settings']['attribution'] = attribution
+        return env
+
+    # The case for NULL in new_profile and NOT NULL in main_summary is
+    # explicitly not tested here because it is not a well defined behavior.
```

Comment:
I'd like to leave the case documented, but I don't think breaking the case is necessarily a bad thing.

In order to avoid double counting clients, the attributes from the first ping of a client taken as a representative.

Essentially

``` sql
select
    attribution, 
    count(client_id) as n_profiles
from (
    select
        client_id,
        attribution,
        row_number() over (order by timestamp desc) as rank
    from main_summary
)
where rank = 1
group by 1
```
as opposed to
```sql
select
    attribution,
    count(distinct client_id) as n_profiles
from main_summary
group by 1
```

Actually after writing that out, I guess it's a good idea to document the quirk in the test. The second query is only viable when you can avoid double counting (i.e. HLL variable).